pyramus/src/main/java/fi/otavanopisto/pyramus/webhooks/WebhookEventListeners.java

```diff
@@ -0,0 +1,229 @@
+  /* Person */
+
+  public void onPersonCreated(@Observes(during=TransactionPhase.AFTER_SUCCESS) PersonCreatedEvent event) {
+    webhooks.sendWebhookNotification(new WebhookPersonCreatePayload(event.getPersonId()));
+  }
+
+  public void onPersonUpdated(@Observes(during=TransactionPhase.AFTER_SUCCESS) PersonUpdatedEvent event) {
+    webhooks.sendWebhookNotification(new WebhookPersonUpdatePayload(event.getPersonId()));
+  }
+
+  public void onPersonArchived(@Observes(during=TransactionPhase.AFTER_SUCCESS) PersonArchivedEvent event) {
+    webhooks.sendWebhookNotification(new WebhookPersonArchivePayload(event.getPersonId()));
+  }
+
+  /* Student */
+  
+  public void onStudentCreated(@Observes(during=TransactionPhase.AFTER_SUCCESS) StudentCreatedEvent event) {
+    webhooks.sendWebhookNotification(new WebhookStudentCreatePayload(event.getStudentId()));
+  }
+  
+  public synchronized void onStudentUpdatedBeforeCompletion(@Observes(during=TransactionPhase.BEFORE_COMPLETION) StudentUpdatedEvent event) {
+    datas.addUpdatedStudentId(event.getStudentId());
+  }
+
+  public synchronized void onStudentUpdatedAfterSuccess(@Observes(during=TransactionPhase.AFTER_SUCCESS) StudentUpdatedEvent event) {
+    List<Long> updatedStudentIds = datas.retrieveUpdatedStudentIds();
+    
+    for (Long updatedStudentId : updatedStudentIds) {
+      webhooks.sendWebhookNotification(new WebhookStudentUpdatePayload(updatedStudentId));
+    }
+  }
+
+  public void onStudentArchived(@Observes(during=TransactionPhase.AFTER_SUCCESS) StudentArchivedEvent event) {
+    webhooks.sendWebhookNotification(new WebhookStudentArchivePayload(event.getStudentId()));
+  }
+  
+  /* Course Staff Member */
+  
+  public void onCourseStaffMemberCreated(@Observes(during=TransactionPhase.AFTER_SUCCESS) CourseStaffMemberCreatedEvent event) {
+    webhooks.sendWebhookNotification(new WebhookCourseStaffMemberCreatePayload(event.getCourseStaffMemberId(), event.getCourseId(), event.getStaffMemberId()));
+  }
+  
+  public synchronized void onCourseStaffMemberUpdatedBeforeCompletion(@Observes(during=TransactionPhase.BEFORE_COMPLETION) CourseStaffMemberUpdatedEvent event) {
+    datas.addUpdatedCourseStaffMember(event.getCourseStaffMemberId(), event.getCourseId(), event.getStaffMemberId());
+  }
+
+  public synchronized void onCourseStaffMemberUpdatedAfterSuccess(@Observes(during=TransactionPhase.AFTER_SUCCESS) CourseStaffMemberUpdatedEvent event) {
+    List<Long> updatedCourseStaffMemberIds = datas.retrieveUpdatedCourseStaffMemberIds();
+    
+    for (Long updatedCourseStaffMemberId : updatedCourseStaffMemberIds) {
+      Long courseId = datas.getCourseStaffMemberCourseId(updatedCourseStaffMemberId);
+      Long staffMemberId = datas.getCourseStaffMemberStaffMemberId(updatedCourseStaffMemberId);
+      webhooks.sendWebhookNotification(new WebhookCourseStaffMemberUpdatePayload(updatedCourseStaffMemberId, courseId, staffMemberId));
+    }
+    
+    datas.clearUpdatedCourseStaffMemberIds();
```

Comment:
What should happen if a webhook notification fails? Should this be cleared partially, totally, or not at all?

Reply:
This is old code as-is. There is a separate issue where these should be covered: https://github.com/otavanopisto/pyramus/issues/536.